src/js/pagination/paging-collection.js

```diff
@@ -83,7 +83,16 @@
              * backbone.paginator.
              */
             parseState: function (response, queryParams, state, options) {
-                return {totalRecords: response[0].count, totalPages: response[0].num_pages};
+                var newState = _.clone(state);
+                newState.totalRecords = response[0].count;
+                newState.totalPages = response[0].num_pages;
+
+                // Remove already processed response
+                delete response.count;
+                delete response.num_pages;
+
+                // Update the state with response
+                return PageableCollection.prototype.parseState.call(this, response, queryParams, newState, options);
             },
```

Comment:
"This method MUST return a new state object instead of directly modifying the PageableCollection#state object. The behavior of directly modifying PageableCollection#state is undefined."

@dan-f Well what this statement tries to convey is that we must return a state object instead of just  modifying the state and returning nothing in which case state is undefined. So this state that is being passed to this function is being passed a copy not the actual state object. So it's safe to modify and return the state, parent is already doing a clone of the state. To be extra safe I have added a new statement to clone the passed state.